doc/guides/west/workspaces.rst

```diff
@@ -0,0 +1,309 @@
+
+Fetching via HTTPS
+==================
+
+On Windows when fetching from GitHub, recent versions of Git prompt you for
+your GitHub password in a graphical window once, then store it for future use
+(in a default installation). Passwordless fetching from GitHub should therefore
+work "out of the box" on Windows after you have done it once.
+
+In general, you can store your credentials on disk using the "store" git
+credential helper. See the `git-credential-store`_ manual page for details.
+
+To use this helper for all the repositories in your workspace, run:
+
+.. code-block:: shell
+
+   west forall -c "git config credential.helper store"
+
+To use this helper on just the projects ``foo`` and ``bar``, run:
+
+.. code-block:: shell
+
+   west forall -c "git config credential.helper store" foo bar
+
+To use this helper by default on your computer, run:
+
+.. code-block:: shell
+
+   git config --global credential.helper store
+
+On GitHub, you can set up a `personal access token`_ to use in place of your
+account password. (This may be required if your account has two-factor
+authentication enabled, and may be preferable to storing your account password
+in plain text even if two-factor authentication is disabed.)
+
+If you don't want to store any credentials on the file system, you can store
+them in memory temporarily using `git-credential-cache`_ instead.
+
+.. _git-credential-store:
+   https://git-scm.com/docs/git-credential-store#_examples
+.. _git-credential-cache:
+   https://git-scm.com/docs/git-credential-cache
+.. _personal access token:
+   https://docs.github.com/en/github/authenticating-to-github/creating-a-personal-access-token
+
+Fetching via SSH
+================
+
+If your SSH key has no password, fetching should just work. If it does have a
+password, you can avoid entering it manually every time using `ssh-agent`_.
+
+On GitHub, see `Connecting to GitHub with SSH`_ for details on configuration
+and key creation.
+
+.. _ssh-agent:
+   https://www.ssh.com/ssh/agent
+.. _Connecting to GitHub with SSH:
+   https://docs.github.com/en/github/authenticating-to-github/connecting-to-github-with-ssh
+
+Project locations
+*****************
+
+Projects can be located anywhere inside the workspace, but they may not
+"escape" it.
+
+In other words, project repositories need not be located in subdirectories of
+the manifest repository or as immediate subdirectories of the topdir. However,
+projects must have paths inside the workspace.
+
+You may replace a project's repository directory within the workspace with a
+symbolic link to elsewhere on your computer, but west will not do this for you.
+
+.. _west-topologies:
+
+Topologies supported
+********************
+
+The following are example source code topologies supported by west.
+
+- T1: star topology, zephyr is the manifest repository
+- T2: star topology, a Zephyr application is the manifest repository
+- T3: forest topology, freestanding manifest repository
```

Comment:
I finally took the time to understand why these analogies repeatedly failed to help me remember their corresponding organizations.

The main issue for me is that a "topology" is normally something spatial, so I kept looking for _directory_ structures looking like "stars" or "forests". Fail. The immediately previous section is all about directories, sure didn't help.

Once lost there it was very hard for me to see that the "star" is a git _repo_ (barely related to directories) and that what makes this repo "the star" is owning the manifest. Obvious... with hindsight.

It's probably too late now to replace "topology" by something less "spatial" ("organization"?) but there could be at least a (non-)transition sentence that clearly departs this section from the previous, directory-focused section and that defines a bit what "topology" means here. For instance:

```
The following are example source code topologies supported by west. A "topology" refers to which git repository is the manifest versioned in.
- T1: star topology, zephyr is the "star" repository because it holds the manifest
- T2: star topology, a Zephyr application is the manifest repository
```

Reply:
I would like to defer improvements to this section until after 2.5. There is some great parallel work in this arena (https://github.com/zephyrproject-rtos/example-application/) that I want to see play out before attempting to bikeshed this.

In no way am I trying to say it's not worth fixing, though. I find this section confusing too.